src/main/java/com/uber/cadence/internal/common/InternalUtils.java

```diff
@@ -80,6 +84,41 @@ public static TaskList createNormalTaskList(String taskListName) {
     return tl;
   }
 
+  public static long awaitTermination(Shutdownable s, long timeoutMillis) {
+    if (s == null) {
+      return timeoutMillis;
+    }
+    return awaitTermination(
+        timeoutMillis,
+        (t) -> {
+          s.awaitTermination(t, TimeUnit.MILLISECONDS);
+        });
+  }
+
+  public static long awaitTermination(ExecutorService s, long timeoutMillis) {
+    if (s == null) {
+      return timeoutMillis;
+    }
+    return awaitTermination(
+        timeoutMillis,
+        (t) -> {
+          try {
+            s.awaitTermination(t, TimeUnit.MILLISECONDS);
+          } catch (InterruptedException e) {
+          }
+        });
+  }
+
+  public static long awaitTermination(long timeoutMillis, Consumer<Long> toTerminate) {
```

Comment:
You technically do not need the passed in value to the lambda here. So it can just be ```Runnable awaitTermination```. But it's up to you.